torchvision/csrc/io/decoder/decoder.cpp

```diff
@@ -505,23 +485,29 @@ int Decoder::getFrame(size_t workingTimeInMs) {
   // once decode() method gets called and grab some bytes
   // run this method again
   // init package
-  AVPacket avPacket;
-  av_init_packet(&avPacket);
-  avPacket.data = nullptr;
-  avPacket.size = 0;
+  // update 03/22: moving memory management to ffmpeg
+  AVPacket* avPacket;
+  avPacket = av_packet_alloc();
+  if (avPacket == nullptr) {
+    LOG(ERROR) << "decoder as not able to allocate the packet.";
+    return AVERROR_BUFFER_TOO_SMALL;
```

Comment:
In this method, this is not a supported error code (at least according to its doc). We should either update the doc to document this error code or adopt a more fitting enum for this.

Reply:
is this better?